src/daft-io/src/s3_like.rs

```diff
@@ -1274,4 +1506,39 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_bytes_chunker_happy_even_split() {
+        let data = bytes::Bytes::from_static(b"1234567890");
+        let chunks: Vec<Bytes> = BytesChunker::chunked(data.clone(), 5).collect();
+        assert_eq!(chunks.len(), 2);
+        assert_eq!(chunks[0], Bytes::from_static(b"12345"));
+        assert_eq!(chunks[1], Bytes::from_static(b"67890"));
+    }
+
+    #[test]
+    fn test_bytes_chunker_happy_uneven_split() {
+        let data = bytes::Bytes::from_static(b"1234567890");
+        let chunks: Vec<Bytes> = BytesChunker::chunked(data.clone(), 3).collect();
+
+        assert_eq!(chunks.len(), 4);
+        assert_eq!(chunks[0], Bytes::from_static(b"123"));
+        assert_eq!(chunks[1], Bytes::from_static(b"456"));
+        assert_eq!(chunks[2], Bytes::from_static(b"789"));
+        assert_eq!(chunks[3], Bytes::from_static(b"0"));
+    }
+
+    #[test]
+    fn test_bytes_chunker_empty() {
```

Comment:
It's only slightly different, but one more case that would be good to test is chunk size larger than data size